test/hotspot/jtreg/runtime/Monitor/UseObjectMonitorTableTest.java

```diff
@@ -0,0 +1,240 @@
+                if (latch != 0) {
+                    latch--;
+                    if (latch == 0) {
+                        notifyAll();
+                    }
+                }
+            }
+        }
+        static class WaitNotifySyncChannel extends WaitNotifyCountDownLatch {
+            WaitNotifyCountDownLatch object;
+            WaitNotifySyncChannel() { super(0); }
+            synchronized void send(WaitNotifyCountDownLatch object, int count) {
+                await();
+                latch = count;
+                this.object = object;
+                notifyAll();
+            }
+            synchronized WaitNotifyCountDownLatch receive() {
+                while (latch == 0) {
+                    try {
+                        wait();
+                    } catch (InterruptedException e) {
+                        throw new RuntimeException("WaitNotifyTest: Unexpected interrupt", e);
+                    }
+                }
+                countDown();
+                return object;
+            }
+        }
+        synchronized int getCount() {
+            return count;
+        }
+        synchronized void increment() {
+            count++;
+        }
+        public void run() {
+            System.out.println("WaitNotifyTest started.");
+            for (int t = 0; t < THREADS; t++) {
+                TF.newThread(() -> {
+                    for (int i = 0; i < ITERATIONS; i++) {
+                        startLatchChannel.receive().await();
+                        increment();
+                        endLatchChannel.receive().countDown();
+                    }
+                }).start();
+            }
+            for (int i = 0; i < ITERATIONS; i++) {
+                WaitNotifyCountDownLatch startLatch = new WaitNotifyCountDownLatch(1);
+                WaitNotifyCountDownLatch endLatch = new WaitNotifyCountDownLatch(THREADS);
+                int count = getCount();
+                if (count != i * THREADS) {
+                    throw new RuntimeException("WaitNotifyTest: Invalid Count " + count +
+                                               " pre-iteration " + i);
+                }
+                startLatchChannel.send(startLatch, 10);
+                startLatch.countDown();
+                endLatchChannel.send(endLatch, 10);
+                endLatch.await();
+            }
+            int count = getCount();
+            if (count != ITERATIONS * THREADS) {
+                throw new RuntimeException("WaitNotifyTest: Invalid Count " + count);
+            }
+            System.out.println("WaitNotifyTest passed.");
+        }
+    }
+
+    static class RandomDepthTest implements Runnable {
+        static final int THREADS = 10;
+        static final int ITERATIONS = 10_000;
+        static final int MAX_DEPTH = 20;
+        static final int MAX_RECURSION_COUNT = 10;
+        static final double RECURSION_CHANCE = .25;
+        final Random random = new Random();
```

Comment:
The test should output a seed value so that the user knows what
random seed value was in use if/when the test fails. Also add a way
to specify the seed value from the command line for reproducibility.